arangod/Transaction/Methods.cpp

```diff
@@ -642,7 +642,11 @@ std::pair<bool, bool> transaction::Methods::findIndexHandleForAndNode(
     if (!sortCondition->isEmpty()) {
       // only take into account the costs for sorting if there is actually
       // something to sort
-      totalCost += sortCost;
+      if (supportsSort) {
+        totalCost += sortCost;
+      } else {
+        totalCost += estimatedItems * std::log2(static_cast<double>(estimatedItems));
```

Comment:
is this always the correct estimation ?

Reply:
It is n * log(n) at least.
However, the optimizer may err about the number of documents to be returned by index after filtering. It simply uses either the selectivityEstimate here (if all equality lookups) or some heuristics that depend on the operators and number of index attributes covered by the filter condition.
@graetzer : Any ideas for improving these estimates?